app/controllers/fhir/r4/api_controller.rb

```diff
@@ -174,7 +174,10 @@ def create
       resource.jurisdiction = resource.creator.jurisdiction
 
       # Generate a submission token for the new monitoree
-      resource.submission_token = SecureRandom.hex(20) # 160 bits
+      loop do
```

Comment:
This code should be encapsulated in a method on the `Patient` model or something, since it's used in multiple places.

Reply:
Addressed here: https://github.com/SaraAlert/SaraAlert/pull/446/commits/38dc5f58019d8aacda9df4952d3b9152de1997fb